### PR TITLE
NP-282: Begin adding config value to validation service and tests, additionally add log supresser to reduce useless console output during tests

### PR DESCRIPTION
**Pull request by Reece-Carruthers**

### CODE REVIEW

**Comment by Reece-Carruthers** [comment]
image

image

File: it/test/uk/gov/hmrc/uknwauthcheckerapi/BaseISpec.scala [hunk 1]
```diff
@@ -44,13 +43,17 @@ class BaseISpec extends PlaySpec with GuiceOneServerPerSuite with WireMockISpec
...
+
   override lazy val app: Application = GuiceApplicationBuilder()
     .configure(additionalAppConfig)
     .build()
+
   protected lazy val authorisationsUrl = s"http://localhost:$port/authorisations"
   protected lazy val appConfig: AppConfig = injected[AppConfig]
   private lazy val wsClient:    WSClient  = injected[WSClient]
 
+  override protected lazy val minMaxValues: MinMaxValues = injected[MinMaxValues]
```
Comment (TheDanielDoyle) [inline it/test/uk/gov/hmrc/uknwauthcheckerapi/BaseISpec.scala:55]: I don't see a point in this overriding a base value. MinMaxValues is a class, and TestData is generated data. We're not generating MinMaxValues, so we can just define it in the BaseISpec if we need it.

File: test/uk/gov/hmrc/uknwauthcheckerapi/controllers/BaseSpec.scala [hunk 2]
```diff
@@ -76,6 +77,7 @@ class BaseSpec
   protected val actorSystem:                            ActorSystem                         = ActorSystem(TestConstants.actorName)
   protected lazy val appConfig:                         AppConfig                           = injected[AppConfig]
   protected lazy val config:                            Config                              = injected[Config]
+  override protected lazy val minMaxValues:             MinMaxValues                        = injected[MinMaxValues]
```
Comment (TheDanielDoyle) [inline test/uk/gov/hmrc/uknwauthcheckerapi/controllers/BaseSpec.scala]: Please can you reorder where this is. It's alphabetical

File: test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala [hunk 3]
```diff
@@ -88,41 +91,112 @@ class ValidationServiceSpec extends BaseSpec {
...
+          val json = Json.toJson(tooManyEorisRequest.request)
 
-        val expectedResponse =
-          ValidationDataRetrievalError(
-            JsError(
-              Seq((JsPath \ JsonPaths.eoris, Seq(JsonValidationError(ApiErrorMessages.invalidEoriCount))))
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq((JsPath \ JsonPaths.eoris, Seq(JsonValidationError("The request payload must contain between 1 and 3000 EORI entries"))))
```
Comment (TheDanielDoyle) [inline test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala]: Can you utilise the ApiErrorMessages. We don't want literals where we can avoid it.

File: test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala [hunk 4]
```diff
@@ -88,41 +91,112 @@ class ValidationServiceSpec extends BaseSpec {
...
+      }
+
+      "when AuthorisationRequest has no Eoris" in {
+        forAll { (noEorisRequest: NoEorisAuthorisationRequest) =>
+          val json = Json.toJson(noEorisRequest.request)
+
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq((JsPath \ JsonPaths.eoris, Seq(JsonValidationError("The request payload must contain between 1 and 3000 EORI entries"))))
```
Comment (TheDanielDoyle) [inline test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala]: Can you utilise the ApiErrorMessages. We don't want literals where we can avoid it.

File: test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala [hunk 5]
```diff
@@ -88,41 +91,112 @@ class ValidationServiceSpec extends BaseSpec {
...
+        forAll { (tooManyEorisRequest: TooManyEorisAuthorisationRequest) =>
+          val json = Json.toJson(tooManyEorisRequest.request)
+
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq(
+                  (
+                    JsPath \ JsonPaths.eoris,
+                    Seq(JsonValidationError(s"The request payload must contain between 1 and $maxEori EORI entries"))
```
Comment (TheDanielDoyle) [inline test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala]: Can you utilise the ApiErrorMessages. We don't want literals where we can avoid it.

File: test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala [hunk 6]
```diff
@@ -88,41 +91,112 @@ class ValidationServiceSpec extends BaseSpec {
...
+        forAll { (noEorisRequest: NoEorisAuthorisationRequest) =>
+          val json = Json.toJson(noEorisRequest.request)
+
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq(
+                  (
+                    JsPath \ JsonPaths.eoris,
+                    Seq(JsonValidationError(s"The request payload must contain between 1 and $maxEori EORI entries"))
```
Comment (TheDanielDoyle) [inline test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala]: Can you utilise the ApiErrorMessages. We don't want literals where we can avoid it.

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**